### PR TITLE
fix httpclient.lastModified bad pattern in parsing

### LINKED ISSUE #12692: unhandled exception: 'Z' is not a valid pattern in httoclient.lastModified
Reproducible with the following code:

```nim
import httpclient
discard lastModified(new Response)
```

When compiling, it will return the following error: `Error: unhandled exception: 'Z' is not a valid pattern [TimeFormatParseError]`

The error comes from parsing, as you can see running the following snippet:
```nim
import times
discard parse("12 Feb 2019", "dd MMM yyyy")
discard parse("12 Feb 2019 Z", "dd MMM yyyy Z")  # will crash at compilation
```

```
$ nim -v
Nim Compiler Version 1.1.1 [Linux: amd64]
git hash: a88004114d6bfbef31268c96f4d1fc871e70ab03
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by zaphodef**
fixes #12692

### CODE REVIEW

**Comment by GULPF** [comment]
Please add a test as well so we can see that it actually works :-)

**Comment by zaphodef** [comment]
Should I add one in `tests/stdlib/thttpclient.nim`? It seems disabled since 73c8391fd3f1f8b789348548f33b2d0ee52c6a75.

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -228,7 +228,7 @@ proc lastModified*(response: Response | AsyncResponse): DateTime =
   ## Raises a ``ValueError`` if the parsing fails or the value is not a correctly
   ## formatted time.
   var lastModifiedHeader = response.headers.getOrDefault("last-modified")
-  result = parse(lastModifiedHeader, "dd, dd MMM yyyy HH:mm:ss Z")
+  result = parse(lastModifiedHeader, "ddd, dd MMM yyyy HH:mm:ss 'GMT'", utc())
```
Comment (dom96) [inline lib/pure/httpclient.nim]: You're adding a default value here, right? That makes the above doc comment outdated as far as I can see.

Comment (zaphodef) [reply lib/pure/httpclient.nim]: `utc()` is the timezone, as we read a GMT time

**Comment by zaphodef** [comment]
[Mozilla documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified) describes the format, and relies on [RFC 7232](https://tools.ietf.org/html/rfc7232#section-2.2), which relies on [RFC 7231](https://tools.ietf.org/html/rfc7231#section-7.1.1.1)

File: lib/pure/httpclient.nim (hunk 1 above)
Comment (narimiran) [reply lib/pure/httpclient.nim]: Btw, why do we use `GMT` and not `UTC`?

Comment (GULPF) [reply lib/pure/httpclient.nim]: `GMT` is just a static part of the format as specified in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified.

**Pull request merged**